protocol/wasmbinding/msg_plugin.go

```diff
@@ -140,3 +197,173 @@ func parseAddress(addr string) (sdk.AccAddress, error) {
 	}
 	return parsed, nil
 }
+
+func (m *CustomMessenger) placeOrder(
+	ctx sdk.Context,
+	contractAddr sdk.AccAddress,
+	placeOrder *bindings.PlaceOrder,
+) ([]sdk.Event, [][]byte, error) {
+	if placeOrder == nil {
+		return nil, nil, wasmvmtypes.InvalidRequest{Err: "create transfer null transfer"}
+	}
+
+	address, err := parseAddress(placeOrder.Order.OrderId.SubaccountId.Owner)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	number := placeOrder.Order.OrderId.SubaccountId.Number
+
+	order := clobtypes.Order{
+		OrderId: clobtypes.OrderId{
+			SubaccountId: satypes.SubaccountId{
+				Owner:  address.String(),
+				Number: number,
+			},
+			ClientId:   placeOrder.Order.OrderId.ClientId,
+			OrderFlags: placeOrder.Order.OrderId.OrderFlags,
+			ClobPairId: placeOrder.Order.OrderId.ClobPairId,
+		},
+		Side:                            clobtypes.Order_Side(placeOrder.Order.Side),
+		Quantums:                        placeOrder.Order.Quantums,
+		Subticks:                        placeOrder.Order.Subticks,
+		TimeInForce:                     clobtypes.Order_TimeInForce(placeOrder.Order.TimeInForce),
+		ReduceOnly:                      placeOrder.Order.ReduceOnly,
+		ClientMetadata:                  placeOrder.Order.ClientMetadata,
+		ConditionType:                   clobtypes.Order_ConditionType(placeOrder.Order.ConditionType),
+		ConditionalOrderTriggerSubticks: placeOrder.Order.ConditionalOrderTriggerSubticks,
+	}
+
+	if placeOrder.Order.OrderId.OrderFlags == clobtypes.OrderIdFlags_ShortTerm {
+		order.GoodTilOneof = &clobtypes.Order_GoodTilBlock{
+			GoodTilBlock: placeOrder.Order.GoodTilBlock,
+		}
+		// throw error for short term contracts since they are not supported
+		return nil, nil, errors.New("short term orders are not supported for contracts")
+	} else {
+		order.GoodTilOneof = &clobtypes.Order_GoodTilBlockTime{
+			GoodTilBlockTime: lib.MustConvertIntegerToUint32(1705794828),
```

Comment:
Shouldn't you use the good til block / good til block from the `placeOrder` message rather than this hardcoded value (used in the hackathon branch for a demo only).

Reply:
you are right, i missed this